tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -403,6 +408,56 @@ std::vector<std::unique_ptr<TChain>> MakeFriends(const ROOT::TreeUtils::RFriendI
    return friends;
 }
 
+std::vector<std::string> ExpandGlob(TString glob)
+{
+   TString directory;
+
+   // TODO: Also implement this pattern in some other free function
+   // This is practically Python's os.path.split
+   Int_t slashpos = glob.Last('/');
+   if (slashpos >= 0) {
+      directory = glob(0, slashpos); // Copy the directory name
+      glob.Remove(0, slashpos + 1);  // and remove it from glob
+   } else {
+      directory = gSystem->UnixPathName(gSystem->WorkingDirectory());
+   }
+
+   const char *epath = gSystem->ExpandPathName(directory.Data());
+   void *dir = gSystem->OpenDirectory(epath);
+   delete[] epath;
+
+   if (dir) {
+      // create a TList to store the file names (not yet sorted)
+      TList l;
+      TRegexp re(glob, true);
+      const char *file;
+      TString s;
+      while ((file = gSystem->GetDirEntry(dir))) {
+         if (!strcmp(file, ".") || !strcmp(file, ".."))
+            continue;
+         s = file;
+         if ((glob != file) && s.Index(re) == kNPOS)
+            continue;
+         // Using "/" as separator here as it was done in TChain::Add
+         // In principle this should be using the appropriate platform separator
+         l.Add(new TObjString((directory + "/" + file).Data()));
```

Comment:
I think this is leaking the TObjStrings. You have to explicitly set `l.SetOwner()`

Reply:
Good catch, I just copied it over from the previous function so probably there was a leak, unless `l.Delete()` also took care of deleting the strings irrespective of the fact there was no call to `SetOwner` before

Reply:
>  unless l.Delete() also took care of deleting the strings irrespective of the fact there was no call to SetOwner before

That is correct.  `Delete` will delete.  `Clear` will delete only if it is the owner.  And `~TList` essentially calls `Clear`